specification.markdown

```diff
@@ -173,21 +173,65 @@ An EBML Stream is a file that consists of one or many EBML Documents that are co
+|----------------|----------|------------|
+| name           | Yes      | The official human-readable name of the EBML Element. The value of the name MUST be in the form of an NCName as defined by the [XML Schema specification](http://www.w3.org/TR/1999/REC-xml-names-19990114/#ns-decl). |
+| level          | Yes      | The level notes at what hierarchical depth the EBML Element may occur within an EBML Document. The initial EBML Element of an EBML Document is at level 0 and the Elements that it may contain are at level 1. The level MUST be expressed as an integer; however, the integer may be followed by a '+' symbol to indicate that the EBML Element is valid at any higher level.  |
+| global         | No       | A boolean to express if an EBML Element MUST occur at its defined level or may occur within any Parent EBML Element. |
+| id             | Yes      | The Element ID expressed in hexadecimal notation prefixed by a '0x'. |
+| mandatory      | No       | A boolean to express if the EBML Element MUST occur if the Parent EBML Element is used. If the mandatory attribute is not expressed for that Element then that element is to be considered not mandatory. |
+| multiple       | No       | A boolean to express if the EBML Element may occur within its Parent EBML Element more than once. If the multiple attribute is false or the  multiple attribute is not used to define the Element then that EBML Element MUST not occur more than once with that Element's Parent EBML Element. |
+| range          | No       | For Elements which are of numerical types (Unsigned Integer, Signed Integer, Float, and Date) a numerical range may be specified. If specified that the value of the EBML Element MUST be within the defined range inclusively. See the [section of Expressions of range](#expression-of-range) for rules applied to expression of range values. |
+| default        | No       | A default value may be provided. If an Element is mandatory but not written within its Parent EBML Element, then the reader of EBML Document MUST consider the defined default value of the Element. EBML Elements that are Master-elements MUST NOT declare a default value. |
+| type           | Yes      | As defined within the [section on EBML Element Types](#ebml-element-types), the type MUST be set to one of the following values: 'integer' (signed integer), 'uinteger' (unsigned integer), 'float', 'string', 'date', 'utf-8', 'master', or 'binary'. |
+
+The value of the `<element>` shall contain a description that of the meaning and use of the EBML Element.
+
+#### Expression of range
+
+The `range` attribute MUST only be used with EBML Elements that are either `signed integer` or `unsigned integer`. The `range` attribute does not support date or float EBML Elements. The `range` expression may contain whitespace for readability but whitespace within a `range` expression MUST NOT convey meaning. The expression of the `range` MUST adhere to one of the following forms:
```

Comment:
I left it applying to only the integers since that was the case in specdata.xml. I suggest we (eventually) ok this pull request to document the current implementation of EBML Schema (specdata.xml) and then in a later pull request, we'll propose a new form of EBML Schema to moreso align with the methods already in place within XML Schema. In XML Schema this case that would mean expressing a restriction like:
```xml
<xs:restriction base="xs:float">
    <xs:minInclusive value="-1.0"/>
    <xs:maxInclusive value="1.0"/>
</xs:restriction>
```